### PR TITLE
Build key(s) to AttributeValue internally

### PR DESCRIPTION
**Pull request by d2a4u**

### CODE REVIEW

File: awssdk/src/main/scala/DefaultClient.scala [hunk 1]
```diff
@@ -291,23 +334,23 @@ class DefaultClient[F[_]: Concurrent: Timer: RaiseThrowable](
   }
 
   def batchDelete[P: Encoder](
```
Comment (joedarby) [inline awssdk/src/main/scala/DefaultClient.scala:336]: Could we add a batchDelete method that takes the items Iterable[T] and returns F[Unit], to be consistent with batchPut?

Comment (d2a4u) [reply awssdk/src/main/scala/DefaultClient.scala:336]: yeah, I will do that as a separate PR. This one is large enough

File: awssdk/src/main/scala/Client.scala [hunk 2]
```diff
@@ -34,17 +34,17 @@ trait Client[F[_]] {
   /**
     * Get a single value from a table by partition key P.
     */
-  def get[U: Decoder, P: Encoder](
-    tableName: String,
+  def get[P: Encoder, U: Decoder](
+    table: Table,
```
Comment (joedarby) [inline awssdk/src/main/scala/Client.scala:38]: The user needs to create a Table now which can be a bit cumbersome:

```      
val partitionKey = Key("id", ScalarAttributeType.S)
val table = Table(tableName, partitionKey, None)
```

In particular the atrributeType of the partition key might not be immediately obvious. Some helper method that assumes it is String perhaps?

File: awssdk/src/main/scala/Client.scala [hunk 3]
```diff
@@ -257,7 +257,7 @@ trait Client[F[_]] {
     * Return Unit (ReturnValue.NONE).
     */
   def update[P: Encoder, S: Encoder](
-    tableName: String,
+    table: Table,
```
Comment (joedarby) [inline awssdk/src/main/scala/Client.scala:260]: Worth having a `put` method that takes a `Table` as well, just for consistency?

**Review by joedarby — APPROVED** [review]
LGTM, just a couple of minor suggestions.

File: awssdk/src/main/scala/Client.scala (hunk 2 above)
Comment (d2a4u) [reply awssdk/src/main/scala/Client.scala:38]: It is a good point, the current implementation uses `Table` for the sake of reusing model but it doesn't check that the `Encoder` for the partition key/sort key matches the type defined by the `Table`. Also, currently there is nothing stopping the user to call `get[P, S, U]` on `Table` that has no sort key. I have some ideas to build a higher API on top of `Table` so will not address this now.

**Pull request merged**